cmd/auth/login.go

```diff
@@ -245,3 +236,34 @@ func getProfileName(authArguments *auth.AuthArguments) string {
 	split := strings.Split(host, ".")
 	return split[0]
 }
+
+func getClusterID(ctx context.Context, profileName string, cfg *config.Config, configureCluster bool, configPath string) (string, error) {
```

Comment:
This function is quite overloaded - it reads the config, asks user input and changes its behaviour based on configureCluster flag.

From the name getClusterID I'd guess that it only reads the cluster_id from the config and does nothing else.

I'd keep configureCluster/NewWorkspaceClient/AskForCluster parts in the RunE method where they were previously and simplified this helper to only read the config unconditionally and do nothing else.